src/utils/random_number_generators.cpp

```diff
@@ -32,32 +32,50 @@
 #include <thread>
 
 namespace {
+#if not defined(LBANN_DETERMINISTIC) && defined(_OPENMP)
 #ifdef __ICC
 lbann::rng_gen generator;
 #pragma omp threadprivate(generator)
```

Comment:
shouldn't this line be removed as well?

Reply:
@tbennun They actually need to be renamed to `OMP_generator`